source/agora/consensus/protocol/Nominator.d

```diff
@@ -280,7 +280,33 @@ extern(D):
     public void setQuorumConfig (ref const(NodeID) node_id,
         const(QuorumConfig)[NodeID] quorums) nothrow @safe
     {
-        assert(!this.is_nominating);
+        // If a nomination is running, we determine the nomination is not
+        // fatal only when the slot id is for the next height. In that case,
+        // we stop the nomination and set the new quorum set.
+        if(this.is_nominating)
+        {
+            const height = this.ledger.getBlockHeight();
+            const slot_idx = height.value + 1;
+            try
+            {
+                () @trusted {
+                    if (this.scp.getHighSlotIndex() != slot_idx)
+                    {
+                        log.fatal("Invalid nomination for slot #{} at height #{}",
+                            slot_idx, height);
```

Comment:
```suggestion
                        log.fatal("Invalid call to setQuorumConfig for SCP slot #{} at ledger height #{}",
                            this.scp.getHighSlotIndex(), height);
```

Reply:
Thanks for the suggestion and will update.